### PR TITLE
github: Improve issue templates to speed up issue triage

### PR DESCRIPTION
**Pull request by serathius**
Add feature request and test flake template to stop users from using wrong or empty template.
Redirect questions to discuss page to remove load from maintainers. 
Remove distributors applications as they are not used at all.

cc @ahrtr @ptabor @spzala

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature-request.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Feature request
```
Comment (spzala) [inline .github/ISSUE_TEMPLATE/feature-request.yml:1]: Nit, suggest to use `Feature Request` to be consistent with other templates (e..g. `Bug Report`, `Flaking Test`)

File: .github/ISSUE_TEMPLATE/feature-request.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    label: What would you like to be added?
+  validations:
+    required: true
+
+- type: textarea
+  id: rationale
+  attributes:
+    label: Why is this needed?
+  validations:
+    required: true
```
Comment (spzala) [inline .github/ISSUE_TEMPLATE/feature-request.yml:18]: Nit, please address GH formatting need `no newline at end of file` (here and other new templates added)

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+contact_links:
```
Comment (spzala) [inline .github/ISSUE_TEMPLATE/config.yml:1]: Curios, did you mean to name the template file `config.yml` ? or good to change to something like `etcd-questions.yml` ?

**Review by spzala — COMMENTED** [review]
@serathius thanks, lgtm but few nits inline.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (serathius) [reply .github/ISSUE_TEMPLATE/config.yml:1]: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser

File: .github/ISSUE_TEMPLATE/feature-request.yml (hunk 1 above)
Comment (serathius) [reply .github/ISSUE_TEMPLATE/feature-request.yml:1]: Could you provide more details about what parts of templates are not consistent?

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (spzala) [reply .github/ISSUE_TEMPLATE/config.yml:2]: Thanks!

File: .github/ISSUE_TEMPLATE/feature-request.yml (hunk 1 above)
Comment (spzala) [reply .github/ISSUE_TEMPLATE/feature-request.yml:1]: It's nit as I said in how the name is mentioned :) : 
`name: Bug Report` in the bug-report.yml
`name: Flaking Test` in the test-flake.yml 
But here in the feature-request.yml it's `name: Feature request`   (i.e. `request` vs `Request`)

Comment (spzala) [reply .github/ISSUE_TEMPLATE/feature-request.yml:1]: If you are making other changes, good to update here too.

**Review by spzala — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: false
```
Comment (ahrtr) [inline .github/ISSUE_TEMPLATE/config.yml:1]: Disabling the blank issue might be controversial, because it forces all users to use the templates and loose the flexibility. I understand that you may be intentionally doing this, but let's revisit this after we get more feedback from the users/community.
